modules/shortcodes/wordads.php

```diff
@@ -53,14 +57,41 @@ static function wordads_shortcode_html( $atts, $content = '' ) {
 			return '<div>' . __( 'The WordAds module is not active', 'jetpack' ) . '</div>';
 		}
 
-		$html = '<div class="jetpack-wordad" itemscope itemtype="https://schema.org/WPAdBlock">';
-
-		$html .= '</div>';
-
+		$html = '<div class="jetpack-wordad" itemscope itemtype="https://schema.org/WPAdBlock"></div>';
 		$html = $wordads->insert_inline_ad( $html );
 
 		return $html;
 	}
+
+	public static function gutenblock_render( $attr ) {
+		global $wordads;
+
+		/** This filter is already documented in modules/wordads/wordads.php `insert_ad()` */
+		if ( is_feed() || apply_filters( 'wordads_inpost_disable', false ) ) {
+			return '';
+		}
+
+		if ( $wordads->option( 'wordads_house' ) ) {
+			return $wordads->get_ad( 'inline', 'house' );
+		}
+
+		$section_id = 0 === $wordads->params->blog_id ? WORDADS_API_TEST_ID : $wordads->params->blog_id . '6'; // 6 is to keep track of gutenblock ads
+		$align = 'center';
+		if ( isset( $attr['align'] ) && in_array( $attr['align'], array( 'left', 'center', 'right' ) ) ) {
+			$align = $attr['align'];
+		}
+		$align = 'align' . $align;
+
+		$format = 'mrec';
+		if ( isset( $attr['format'] ) && in_array( $attr['format'], array_keys( WordAds::$ad_tag_ids ) ) ) {
+			$format = $attr['format'];
+		}
+
+		$height = WordAds::$ad_tag_ids[$format]['height'];
+		$width = WordAds::$ad_tag_ids[$format]['width'];
+		$snippet = $wordads->get_ad_snippet( $section_id, $height, $width, 'inline', 'float:left;margin-right:5px;margin-top:0px;' );
+		return $wordads->get_ad_div( 'inline', $snippet, array( $align ) );
+	}
 }
 
 new Jetpack_WordAds_Shortcode();
```

Comment:
With Gutenberg, we should start to think of shortcodes as legacy. Rather than adding the rendering here in the WordAds shortcode, how about adding this code in another spot? Maybe a new file in the WordAds module?

Reply:
Putting the code here was a recommendation by @enejb, there seems to be some precedent in Jetpack for putting the Gutenberg blocks in the shortcode path, but I'm neutral on the subject.

Reply:
I'd say it makes more sense to be nearer to the module providing its functionality. That's what simple payments does:

https://github.com/Automattic/jetpack/blob/7b7d869dbaa6131e8782431c0f78638cc990ea98/modules/simple-payments/simple-payments.php#L43

If you think this is a more appropriate place you can leave it here.

Reply:
@enejb and I seem to think that it's fine to leave here for now and until the shortcode goes away. At any rate, the refactoring seems best to go into a separate PR if needed.

Reply:
That's fine 🙂